sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -57,191 +55,106 @@
-    codersToRegister.put(Iterable.class, CoderFactories.fromStaticMethods(IterableCoder.class));
-    codersToRegister.put(KV.class, CoderFactories.fromStaticMethods(KvCoder.class));
-    codersToRegister.put(List.class, CoderFactories.fromStaticMethods(ListCoder.class));
-    codersToRegister.put(Long.class, CoderFactories.fromStaticMethods(VarLongCoder.class));
-    codersToRegister.put(Map.class, CoderFactories.fromStaticMethods(MapCoder.class));
-    codersToRegister.put(Set.class, CoderFactories.fromStaticMethods(SetCoder.class));
-    codersToRegister.put(String.class, CoderFactories.fromStaticMethods(StringUtf8Coder.class));
-    codersToRegister.put(TimestampedValue.class,
-        CoderFactories.fromStaticMethods(TimestampedValue.TimestampedValueCoder.class));
-    codersToRegister.put(Void.class, CoderFactories.fromStaticMethods(VoidCoder.class));
-    codersToRegister.put(byte[].class, CoderFactories.fromStaticMethods(ByteArrayCoder.class));
-    codersToRegister.put(IntervalWindow.class, CoderFactories.forCoder(IntervalWindow.getCoder()));
+    List<CoderFactory> codersToRegister = new ArrayList<>();
+    codersToRegister.add(CoderFactories.fromStaticMethods(Byte.class, ByteCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(BitSet.class, BitSetCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Double.class, DoubleCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Instant.class, InstantCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Integer.class, VarIntCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Iterable.class, IterableCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(KV.class, KvCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(List.class, ListCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Long.class, VarLongCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Map.class, MapCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Set.class, SetCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(String.class, StringUtf8Coder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(
+        TimestampedValue.class, TimestampedValue.TimestampedValueCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(Void.class, VoidCoder.class));
+    codersToRegister.add(CoderFactories.fromStaticMethods(byte[].class, ByteArrayCoder.class));
+    codersToRegister.add(
+        CoderFactories.forCoder(
+            TypeDescriptor.of(IntervalWindow.class), IntervalWindow.getCoder()));
 
     // Enumerate all the CoderRegistrars in a deterministic order, adding all coders to register
-    Set<CoderRegistrar> registrars = Sets.newTreeSet(ObjectsClassComparator.INSTANCE);
+    Set<CoderFactoryRegistrar> registrars = Sets.newTreeSet(ObjectsClassComparator.INSTANCE);
```

Comment:
Using a deterministic order could cause accidental success, leading to mystery breakage later on. Is there another way to regulate duplicate registrations?